packages/extension-link/src/helpers/keydownHandler.ts

```diff
@@ -0,0 +1,24 @@
+import { Plugin, PluginKey, TextSelection } from '@tiptap/pm/state'
+
+export function keydownHandler(): Plugin {
+  return new Plugin({
+    key: new PluginKey('handleKeydownLink'),
+    props: {
+      handleKeyDown: (view, event) => {
+        if (!view.editable) {
+          return false
+        }
+
+        const { selection } = view.state
+
+        if (event.key === 'Escape' && !selection.empty) {
+          view.dispatch(view.state.tr.setSelection(TextSelection.create(view.state.doc, selection.to)))
```

Comment:
Do we need to check if a link is selected here? A bit worried that this key handler will take precedence over other keyhandlers on escape